drizzlepac/devutils/comparison_tools/read_hla/read_hla_catalog.py

```diff
@@ -0,0 +1,156 @@
+    log.setLevel(log_level)
+    cattype = cattype.lower()
+    if cattype not in catlist:
+        raise ValueError("cattype '{}' must be one of {}".format(cattype, " ".join(catlist)))
+    if catformat.lower() not in formatlist:
+        raise ValueError("catformat '{}' must be one of {}".format(catformat, " ".join(formatlist)))
+    # get URL for this catalog
+    params = ['catalog='+cattype, 'format='+catformat]
+    if trim:
+        suffix = "{}phot_trm.cat".format(cattype)
+    else:
+        suffix = "{}phot.cat".format(cattype)
+        params.append('ignoreflag=T')
+    if multiwave:
+        params.append('multiwave=T')
+        f = '_'.join(dataset.split('_')[:5])
+        dataset = '{}_total'.format(f)
+        catname = '{}_multiwave'.format(f)
+    else:
+        catname = dataset
+    params.append('image={}'.format(dataset))
+    params.append('filename={}_{}'.format(catname, suffix))
+    caturl = "{}?{}".format(url, "&".join(params))
+    if catformat == "votable":
+        cat = Table.read(caturl, format=catformat)
+    else:
+        # csv (and other?) formats start with a comment
+        cat = Table.read(caturl, format=catformat, comment='#')
+    if verbose:
+        if multiwave:
+            log.info("Retrieved multiwave {} catalog for {} with {} rows".format(cattype, dataset, len(cat)))
+        else:
+            log.info("Retrieved {} catalog for {} with {} rows".format(cattype, dataset, len(cat)))
+        ncols = 10
+        for i in range(0, len(cat.colnames), ncols):
+            cc = cat.colnames[i:i+ncols]
+            log.info(("{:15} "*len(cc)).format(*cc))
+    if applyomega:
+        omega = omegaxyz.getomegaxyz(dataset)
+        if omega == (0.0, 0.0, 0.0):
+            if verbose:
+                log.info("Omega correction is zero for this visit, no correction applied")
+        else:
+            if verbose:
+                log.info("Applying omega correction to astrometry")
+            # names of RA and Dec columns varies for different catalogs
+            racol, deccol = get_radec_cols(cat)
+            newra, newdec = omegaxyz.applyomegacat(cat[racol], cat[deccol], omega)
+            cat[racol] = newra
+            cat[deccol] = newdec
+    return cat
+
+
+def get_radec_cols(cat):
+
+    """Return names of the RA and Dec columns for this catalog.
+
+    Parameters
+    ----------
+    cat : Astropy Table object
+        catalog to extract RA and Dec columns from
+
+    Returns
+    -------
+    racol : Astropy.table.column object
+        RA column values
+
+    deccol : Astropy.table.column object
+        Dec column values
+    """
+
+    # create a column name dictionary
+    d = dict([(x, 1) for x in cat.colnames])
+    for v in ["ra", "RA", "ALPHA_J2000", "alpha_j2000"]:
+        if v in d:
+            racol = v
+            break
+    else:
+        raise ValueError("RA column not found in", " ".join(cat.colnames))
+    for v in ["dec", "DEC", "Dec", "DELTA_J2000", "delta_j2000"]:
```

Comment:
Same comment as previously stated in another routine (can be addressed later), convert comparisons all to lower- or upper-case.

Reply:
Resolved by commit [9e55ef4](https://github.com/spacetelescope/drizzlepac/pull/548/commits/9e55ef4b4ae79063fdb1250143e3d7a2477372aa)

Reply:
It actually crashes with these changes. Above changes rolled back in commit [7e1e135](https://github.com/spacetelescope/drizzlepac/pull/548/commits/7e1e1350b491e9a3f7125f6a041aed0183547407)